app/src/main/java/org/p2p/wallet/history/ui/detailsbottomsheet/HistoryTransactionDetailsBottomSheetPresenter.kt

```diff
@@ -116,33 +114,49 @@ class HistoryTransactionDetailsBottomSheetPresenter(
 
     private fun parseBurnOrMint(transaction: RpcHistoryTransaction.BurnOrMint) {
         view?.apply {
-            showDate(transaction.date.toDateTimeString())
-
             showTransactionId(transaction.signature)
-            showAddresses(transaction.senderAddress, transaction.destination)
 
             val usdTotal = transaction.getFormattedAmount()
             val total = transaction.getFormattedTotal()
             showAmount(total, usdTotal)
             showFee()
-            showTransferView(transaction.getIcon())
+            showTransferView(transaction.iconUrl, transaction.getIcon())
+            hideSendReceiveTitleAndValue()
         }
     }
 
-    private fun logTransactionClicked(transaction: HistoryTransaction) {
-        when (transaction) {
-            is RpcHistoryTransaction.Swap -> {
-                historyAnalytics.logSwapTransactionClicked(transaction)
+    private fun showStatusAndSignature(transaction: RpcHistoryTransaction?) {
+        val status = transaction?.status ?: return
+        val colorRes: Int
+        val titleRes: Int
+        when (status) {
+            HistoryTransactionStatus.COMPLETED -> {
+                titleRes = R.string.transaction_details_title_succeeded
+                colorRes = if (transaction is RpcHistoryTransaction.Swap) R.color.text_night else R.color.text_mint
             }
-            is RpcHistoryTransaction.Transfer -> {
-                launch {
-                    historyAnalytics.logTransferTransactionClicked(
-                        transaction = transaction,
-                        isRenBtcSessionActive = renBtcInteractor.isUserHasActiveSession()
-                    )
-                }
+            HistoryTransactionStatus.PENDING -> {
+                titleRes = R.string.transaction_details_title_submitted
+                colorRes = R.color.text_sun
             }
-            else -> Unit
+            HistoryTransactionStatus.ERROR -> {
+                titleRes = R.string.transaction_details_title_failed
+                colorRes = R.color.text_rose
+                showErrorState(transaction)
+            }
+        }
+        view?.apply {
```

Comment:
it's not apply, I think, it will be let

Reply:
apply works here
and view?.showStatus ; view?.showTransactionId works too